Use "= default" for constructor and destructor in JSC code

Ahmad-S792 opened the pull request:
#### f61dcc33e889168e8dc6fa25ce9478ca65786a45
<pre>
Use &quot;= default&quot; for constructor and destructor in JSC code

<a href="https://bugs.webkit.org/show_bug.cgi?id=274978">https://bugs.webkit.org/show_bug.cgi?id=274978</a>

Reviewed by Chris Dumez.

This extends our &quot;= default&quot; usage across constructor and destructor in JSC code.

* Source/JavaScriptCore/API/ObjCCallbackFunction.mm:
(CallbackArgument::~CallbackArgument):
* Source/JavaScriptCore/b3/B3ArgumentRegValue.cpp:
(JSC::B3::ArgumentRegValue::~ArgumentRegValue):
* Source/JavaScriptCore/b3/B3AtomicValue.cpp:
(JSC::B3::AtomicValue::~AtomicValue):
* Source/JavaScriptCore/b3/B3BasicBlock.cpp:
(JSC::B3::BasicBlock::~BasicBlock):
* Source/JavaScriptCore/b3/B3BlockInsertionSet.cpp:
(JSC::B3::BlockInsertionSet::~BlockInsertionSet):
* Source/JavaScriptCore/b3/B3CCallValue.cpp:
(JSC::B3::CCallValue::~CCallValue):
* Source/JavaScriptCore/b3/B3CheckSpecial.cpp:
(JSC::B3::CheckSpecial::~CheckSpecial):
* Source/JavaScriptCore/b3/B3CheckValue.cpp:
(JSC::B3::CheckValue::~CheckValue):
* Source/JavaScriptCore/b3/B3Const32Value.cpp:
(JSC::B3::Const32Value::~Const32Value):
* Source/JavaScriptCore/b3/B3Const64Value.cpp:
(JSC::B3::Const64Value::~Const64Value):
* Source/JavaScriptCore/b3/B3ConstDoubleValue.cpp:
(JSC::B3::ConstDoubleValue::~ConstDoubleValue):
* Source/JavaScriptCore/b3/B3ConstFloatValue.cpp:
(JSC::B3::ConstFloatValue::~ConstFloatValue):
* Source/JavaScriptCore/b3/B3ExtractValue.cpp:
(JSC::B3::ExtractValue::~ExtractValue):
* Source/JavaScriptCore/b3/B3FenceValue.cpp:
(JSC::B3::FenceValue::~FenceValue):
* Source/JavaScriptCore/b3/B3MemoryValue.cpp:
(JSC::B3::MemoryValue::~MemoryValue):
* Source/JavaScriptCore/b3/B3PatchpointSpecial.cpp:
(JSC::B3::PatchpointSpecial::PatchpointSpecial):
(JSC::B3::PatchpointSpecial::~PatchpointSpecial):
* Source/JavaScriptCore/b3/B3PatchpointValue.cpp:
(JSC::B3::PatchpointValue::~PatchpointValue):
* Source/JavaScriptCore/b3/B3PhiChildren.cpp:
(JSC::B3::PhiChildren::~PhiChildren):
* Source/JavaScriptCore/b3/B3Procedure.cpp:
(JSC::B3::Procedure::~Procedure):
* Source/JavaScriptCore/b3/B3PureCSE.cpp:
(JSC::B3::PureCSE::PureCSE):
(JSC::B3::PureCSE::~PureCSE):
* Source/JavaScriptCore/b3/B3SSACalculator.cpp:
(JSC::B3::SSACalculator::~SSACalculator):
* Source/JavaScriptCore/b3/B3SlotBaseValue.cpp:
(JSC::B3::SlotBaseValue::~SlotBaseValue):
* Source/JavaScriptCore/b3/B3StackmapSpecial.cpp:
(JSC::B3::StackmapSpecial::StackmapSpecial):
(JSC::B3::StackmapSpecial::~StackmapSpecial):
* Source/JavaScriptCore/b3/B3StackmapValue.cpp:
(JSC::B3::StackmapValue::~StackmapValue):
* Source/JavaScriptCore/b3/B3SwitchValue.cpp:
(JSC::B3::SwitchValue::~SwitchValue):
* Source/JavaScriptCore/b3/B3UpsilonValue.cpp:
(JSC::B3::UpsilonValue::~UpsilonValue):
* Source/JavaScriptCore/b3/B3UseCounts.cpp:
(JSC::B3::UseCounts::~UseCounts):
* Source/JavaScriptCore/b3/B3Variable.cpp:
(JSC::B3::Variable::~Variable):
* Source/JavaScriptCore/b3/B3VariableLiveness.cpp:
(JSC::B3::VariableLiveness::~VariableLiveness):
* Source/JavaScriptCore/b3/B3VariableValue.cpp:
(JSC::B3::VariableValue::~VariableValue):
* Source/JavaScriptCore/b3/B3WasmAddressValue.cpp:
(JSC::B3::WasmAddressValue::~WasmAddressValue):
* Source/JavaScriptCore/b3/B3WasmBoundsCheckValue.cpp:
(JSC::B3::WasmBoundsCheckValue::~WasmBoundsCheckValue):
* Source/JavaScriptCore/b3/air/AirBlockInsertionSet.cpp:
(JSC::B3::Air::BlockInsertionSet::~BlockInsertionSet):
* Source/JavaScriptCore/b3/air/AirCCallSpecial.cpp:
(JSC::B3::Air::CCallSpecial::~CCallSpecial):
* Source/JavaScriptCore/b3/air/AirCode.cpp:
(JSC::B3::Air::Code::~Code):
* Source/JavaScriptCore/b3/air/AirPrintSpecial.cpp:
(JSC::B3::Air::PrintSpecial::~PrintSpecial):
* Source/JavaScriptCore/b3/air/AirSpecial.cpp:
(JSC::B3::Air::Special::Special):
(JSC::B3::Air::Special::~Special):
* Source/JavaScriptCore/b3/air/AirTmpWidth.cpp:
(JSC::B3::Air::TmpWidth::~TmpWidth):
* Source/JavaScriptCore/bytecode/CheckPrivateBrandVariant.cpp:
(JSC::CheckPrivateBrandVariant::~CheckPrivateBrandVariant):
* Source/JavaScriptCore/bytecode/DFGExitProfile.cpp:
(JSC::DFG::ExitProfile::ExitProfile):
(JSC::DFG::ExitProfile::~ExitProfile):
* Source/JavaScriptCore/bytecode/DeferredCompilationCallback.cpp:
(JSC::DeferredCompilationCallback::DeferredCompilationCallback):
(JSC::DeferredCompilationCallback::~DeferredCompilationCallback):
* Source/JavaScriptCore/bytecode/DeleteByVariant.cpp:
(JSC::DeleteByVariant::~DeleteByVariant):
* Source/JavaScriptCore/bytecode/GetByVariant.cpp:
(JSC::GetByVariant::~GetByVariant):
* Source/JavaScriptCore/bytecode/InlineCallFrameSet.cpp:
(JSC::InlineCallFrameSet::InlineCallFrameSet):
(JSC::InlineCallFrameSet::~InlineCallFrameSet):
* Source/JavaScriptCore/bytecode/SetPrivateBrandVariant.cpp:
(JSC::SetPrivateBrandVariant::~SetPrivateBrandVariant):
* Source/JavaScriptCore/bytecode/TrackedReferences.cpp:
(JSC::TrackedReferences::TrackedReferences):
(JSC::TrackedReferences::~TrackedReferences):
* Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp:
(JSC::BytecodeGenerator::~BytecodeGenerator):
* Source/JavaScriptCore/debugger/Debugger.cpp:
(JSC::Debugger::ProfilingClient::~ProfilingClient):

Canonical link: <a href="https://commits.webkit.org/279604@main">https://commits.webkit.org/279604@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6551b30a7a959343406f460e7ee9d39d346d536e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53901 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/33271 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/6434 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/57180 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/4625 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/40777 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/4502 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/43646 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/3048 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/55998 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/31516 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/46627 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/24804 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/28319 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/3946 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/2778 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/47260 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/49983 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/4147 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/58773 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/53413 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/29075 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/4223 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/51063 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/30266 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/46767 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/50396 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11751 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/31213 "Built successfully") | | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/65714 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30043 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/12506 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

vitorroriz commented on Source/JavaScriptCore/b3/B3ArgumentRegValue.cpp:
```diff
@@ -30,9 +30,7 @@
-ArgumentRegValue::~ArgumentRegValue()
-{
-}
+ArgumentRegValue::~ArgumentRegValue() = default;
```
Shouldn't we use default directly at the header files?

Ahmad-S792 replied:
I have seen `= default` used in WebKit code in this way, can you share example, so I can approach it your suggested way.

vitorroriz replied:
Example: https://github.com/WebKit/WebKit/blob/78d8c59ed298b1c74189daf439a9104dbb7b2e2c/Source/WTF/wtf/HashMap.h#L89

cdumez replied:
Ahmad, please do not apply this suggestion. We do not want to inline constructors / destructors in this patch, this would not be identical to the pre-existing code and could have performance implications.

cdumez reviewed (approved):
LGTM

Commit:
Use "= default" for constructor and destructor in JSC code

https://bugs.webkit.org/show_bug.cgi?id=274978

Reviewed by Chris Dumez.

This extends our "= default" usage across constructor and destructor in JSC code.

* Source/JavaScriptCore/API/ObjCCallbackFunction.mm:
(CallbackArgument::~CallbackArgument):
* Source/JavaScriptCore/b3/B3ArgumentRegValue.cpp:
(JSC::B3::ArgumentRegValue::~ArgumentRegValue):
* Source/JavaScriptCore/b3/B3AtomicValue.cpp:
(JSC::B3::AtomicValue::~AtomicValue):
* Source/JavaScriptCore/b3/B3BasicBlock.cpp:
(JSC::B3::BasicBlock::~BasicBlock):
* Source/JavaScriptCore/b3/B3BlockInsertionSet.cpp:
(JSC::B3::BlockInsertionSet::~BlockInsertionSet):
* Source/JavaScriptCore/b3/B3CCallValue.cpp:
(JSC::B3::CCallValue::~CCallValue):
* Source/JavaScriptCore/b3/B3CheckSpecial.cpp:
(JSC::B3::CheckSpecial::~CheckSpecial):
* Source/JavaScriptCore/b3/B3CheckValue.cpp:
(JSC::B3::CheckValue::~CheckValue):
* Source/JavaScriptCore/b3/B3Const32Value.cpp:
(JSC::B3::Const32Value::~Const32Value):
* Source/JavaScriptCore/b3/B3Const64Value.cpp:
(JSC::B3::Const64Value::~Const64Value):
* Source/JavaScriptCore/b3/B3ConstDoubleValue.cpp:
(JSC::B3::ConstDoubleValue::~ConstDoubleValue):
* Source/JavaScriptCore/b3/B3ConstFloatValue.cpp:
(JSC::B3::ConstFloatValue::~ConstFloatValue):
* Source/JavaScriptCore/b3/B3ExtractValue.cpp:
(JSC::B3::ExtractValue::~ExtractValue):
* Source/JavaScriptCore/b3/B3FenceValue.cpp:
(JSC::B3::FenceValue::~FenceValue):
* Source/JavaScriptCore/b3/B3MemoryValue.cpp:
(JSC::B3::MemoryValue::~MemoryValue):
* Source/JavaScriptCore/b3/B3PatchpointSpecial.cpp:
(JSC::B3::PatchpointSpecial::PatchpointSpecial):
(JSC::B3::PatchpointSpecial::~PatchpointSpecial):
* Source/JavaScriptCore/b3/B3PatchpointValue.cpp:
(JSC::B3::PatchpointValue::~PatchpointValue):
* Source/JavaScriptCore/b3/B3PhiChildren.cpp:
(JSC::B3::PhiChildren::~PhiChildren):
* Source/JavaScriptCore/b3/B3Procedure.cpp:
(JSC::B3::Procedure::~Procedure):
* Source/JavaScriptCore/b3/B3PureCSE.cpp:
(JSC::B3::PureCSE::PureCSE):
(JSC::B3::PureCSE::~PureCSE):
* Source/JavaScriptCore/b3/B3SSACalculator.cpp:
(JSC::B3::SSACalculator::~SSACalculator):
* Source/JavaScriptCore/b3/B3SlotBaseValue.cpp:
(JSC::B3::SlotBaseValue::~SlotBaseValue):
* Source/JavaScriptCore/b3/B3StackmapSpecial.cpp:
(JSC::B3::StackmapSpecial::StackmapSpecial):
(JSC::B3::StackmapSpecial::~StackmapSpecial):
* Source/JavaScriptCore/b3/B3StackmapValue.cpp:
(JSC::B3::StackmapValue::~StackmapValue):
* Source/JavaScriptCore/b3/B3SwitchValue.cpp:
(JSC::B3::SwitchValue::~SwitchValue):
* Source/JavaScriptCore/b3/B3UpsilonValue.cpp:
(JSC::B3::UpsilonValue::~UpsilonValue):
* Source/JavaScriptCore/b3/B3UseCounts.cpp:
(JSC::B3::UseCounts::~UseCounts):
* Source/JavaScriptCore/b3/B3Variable.cpp:
(JSC::B3::Variable::~Variable):
* Source/JavaScriptCore/b3/B3VariableLiveness.cpp:
(JSC::B3::VariableLiveness::~VariableLiveness):
* Source/JavaScriptCore/b3/B3VariableValue.cpp:
(JSC::B3::VariableValue::~VariableValue):
* Source/JavaScriptCore/b3/B3WasmAddressValue.cpp:
(JSC::B3::WasmAddressValue::~WasmAddressValue):
* Source/JavaScriptCore/b3/B3WasmBoundsCheckValue.cpp:
(JSC::B3::WasmBoundsCheckValue::~WasmBoundsCheckValue):
* Source/JavaScriptCore/b3/air/AirBlockInsertionSet.cpp:
(JSC::B3::Air::BlockInsertionSet::~BlockInsertionSet):
* Source/JavaScriptCore/b3/air/AirCCallSpecial.cpp:
(JSC::B3::Air::CCallSpecial::~CCallSpecial):
* Source/JavaScriptCore/b3/air/AirCode.cpp:
(JSC::B3::Air::Code::~Code):
* Source/JavaScriptCore/b3/air/AirPrintSpecial.cpp:
(JSC::B3::Air::PrintSpecial::~PrintSpecial):
* Source/JavaScriptCore/b3/air/AirSpecial.cpp:
(JSC::B3::Air::Special::Special):
(JSC::B3::Air::Special::~Special):
* Source/JavaScriptCore/b3/air/AirTmpWidth.cpp:
(JSC::B3::Air::TmpWidth::~TmpWidth):
* Source/JavaScriptCore/bytecode/CheckPrivateBrandVariant.cpp:
(JSC::CheckPrivateBrandVariant::~CheckPrivateBrandVariant):
* Source/JavaScriptCore/bytecode/DFGExitProfile.cpp:
(JSC::DFG::ExitProfile::ExitProfile):
(JSC::DFG::ExitProfile::~ExitProfile):
* Source/JavaScriptCore/bytecode/DeferredCompilationCallback.cpp:
(JSC::DeferredCompilationCallback::DeferredCompilationCallback):
(JSC::DeferredCompilationCallback::~DeferredCompilationCallback):
* Source/JavaScriptCore/bytecode/DeleteByVariant.cpp:
(JSC::DeleteByVariant::~DeleteByVariant):
* Source/JavaScriptCore/bytecode/GetByVariant.cpp:
(JSC::GetByVariant::~GetByVariant):
* Source/JavaScriptCore/bytecode/InlineCallFrameSet.cpp:
(JSC::InlineCallFrameSet::InlineCallFrameSet):
(JSC::InlineCallFrameSet::~InlineCallFrameSet):
* Source/JavaScriptCore/bytecode/SetPrivateBrandVariant.cpp:
(JSC::SetPrivateBrandVariant::~SetPrivateBrandVariant):
* Source/JavaScriptCore/bytecode/TrackedReferences.cpp:
(JSC::TrackedReferences::TrackedReferences):
(JSC::TrackedReferences::~TrackedReferences):
* Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp:
(JSC::BytecodeGenerator::~BytecodeGenerator):
* Source/JavaScriptCore/debugger/Debugger.cpp:
(JSC::Debugger::ProfilingClient::~ProfilingClient):

Canonical link: https://commits.webkit.org/279604@main
